main.cpp

```diff
@@ -225,33 +175,70 @@ float FloatType::divide( float lhs, float rhs )
 
 struct DoubleType
 {
-    double add( double lhs, double rhs );
-    double subtract( double lhs, double rhs );
-    double multiply( double lhs, double rhs );
-    double divide( double lhs, double rhs );   
+    double* valuePtr = new double;
+    double& value = *valuePtr;
+
+    DoubleType(double d) : value (d) {}
+    ~DoubleType()
+    {
+        delete valuePtr;
+        valuePtr = nullptr;
+    }
+
+    DoubleType& add(double rhs );
+    DoubleType& add(const FloatType f);
+
+    DoubleType& subtract(double rhs );
+    DoubleType& multiply(double rhs );
+    DoubleType& multiply(const IntType& i);
+    DoubleType& divide(double rhs );  
+    DoubleType& devide(const IntType& i); 
```

Comment:
You need to add 1 `add/subtract/multiply/divide` function for each of the 3 types.
```
 4) write add/subtract/divide/multiply member functions for each type that take your 3 UDTs
        These are in addition to your member functions that take primitives
        for example, IntType::divide(const DoubleType& dt);
        These functions should return the result of calling the function that takes the primitive.
```
- [ ] revise